src/cmds/fs/uniq.c

```diff
@@ -0,0 +1,185 @@
+/**
+ * @file
+ * @brief Filter  adjacent matching lines from FILE, writing to standard output.
+ *
+ * @date 02.07.19
+ * @author Nastya Nizharadze
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <errno.h>
+
+static int number_prefix;
+static int dup_only;
+static int nul_end;
+static int all_dup;
+
+static int max_length;
+static int dup_count;
+
+static void print_usage(void) {
+	printf(
+		"Usage: uniq [OPTION]... [FILE]...\n"
+		"Filter  adjacent matching lines from FILE, "
+		"writing to standard output.\n"
+		"\nOptions:\n\n"
+		"  -c\tprefix lines by the number of occurrences\n"
+		"  -d\tonly print duplicate lines, one for each group\n"
+		"  -D\tprint all duplicate lines\n"
+		"  -z\tline delimiter is NUL, not newline\n"
+		"  -h\tdisplay this help and exit\n"
+	);
+}
+
+static int read_line(char* line_buff, FILE *fp) {
+	if (NULL != fgets(line_buff, max_length + 3, fp)) {
+		int str_size = strlen(line_buff);
+		if (line_buff[str_size - 1] == '\n') {
+			if (1 == nul_end) {
```

Comment:
You can simply write `if (nul_end)`, it's same as `if (nul_end != 0)`

Reply:
Done.